gdx/src/com/badlogic/gdx/utils/ArrayMap.java

```diff
@@ -398,6 +398,42 @@ public void truncate (int newSize) {
 		size = newSize;
 	}
 
+	public int hashCode () {
+		K[] keys = this.keys;
+		V[] values = this.values;
+		int h = 0;
+		for (int i = 0, n = size; i < n; i++) {
+			K key = keys[i];
+			V value = values[i];
+			if (key != null) h += key.hashCode() * 31;
+			if (value != null) h += value.hashCode();

```

Comment:
That's probably a worse implementation, as it results in the same hash for swapped keys/values. `MapEntry` uses that implementation as it's specified in `java.util.Map.Entry`.